reconcile/utils/jump_host.py

```diff
@@ -71,3 +77,19 @@ def get_ssh_base_cmd(self):
             '-o', 'StrictHostKeyChecking=yes',
             '-o', 'UserKnownHostsFile={}'.format(self.known_hosts_file),
             '-i', self.identity_file, '-p', str(self.port), user_host]
+
+    def create_ssh_tunnel(self, local_port, remote_port):
+        key = f'{self.hostname}-{local_port}-{remote_port}'
+        JumpHostSSH.tunnel_lock.acquire()
+        if key not in JumpHostSSH.bastion_tunnel:
+            tunnel = sshtunnel.SSHTunnelForwarder(
+                ssh_address_or_host=self.hostname,
+                ssh_port=self.port,
+                ssh_username=self.user,
+                ssh_pkey=self.identity_file,
+                remote_bind_address=(self.hostname, remote_port),
+                local_bind_address=('localhost', local_port)
+            )
+            tunnel.start()
+            JumpHostSSH.bastion_tunnel[key] = tunnel
+        JumpHostSSH.tunnel_lock.release()
```

Comment:
@apahim I would like to use defer here instead, but my attempts to use it failed.  Should just be:

@defer
def create_ssh_tunnel(...):
...
acquire()
defer(JumpHostSSH.tunnel_lock.release())

Right?

Reply:
`defer()` takes a callable, not an object instance

Reply:
`defer(JumpHostSSH.tunnel_lock.release)` should do it

Reply:
Hrm, using defer results in about a 30+s decrease in time to complete.  I'd prefer to use it, but for now I will leave it as is since a goal of this is to improve performance.  Giving back 30+s is a pretty big chunk.

Reply:
that doesn't sound right :)

Reply:
But agreed, adding 30s is not acceptable, specially when we are fighting for faster executions.

Reply:
I was surprised by the degradation too.  I ran it twice, and the 2nd time was a little better but still about 30s slower than without:
1st run:
real	1m56.568s
user	1m6.170s
sys	0m11.752s

2nd run:
real	1m46.512s
user	1m5.351s
sys	0m11.312s

current (no defer):
real	1m15.839s
user	1m4.597s
sys	0m11.181s